library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -2079,7 +2089,7 @@ private static boolean isSupportedAdaptiveAudioTrack(
           boolean isDefault = (maskedSelectionFlags & C.SELECTION_FLAG_DEFAULT) != 0;
           boolean isForced = (maskedSelectionFlags & C.SELECTION_FLAG_FORCED) != 0;
           int trackScore;
-          int languageScore = getFormatLanguageScore(format, params.preferredTextLanguage);
+          int languageScore = getFormatLanguageScore(format, params.preferredTextLanguage, params.preferredRoleFlags);
```

Comment:
The track score logic in selectTextTrack gets increasingly complicated, especially if we follow the proposal I wrote down above. I think before continuing with this change, it would be best to switch from the score based logic to a comparison based logic similar to the one we use for audio track selection (see AudioTrackScore). This allows to get rid of these seemingly arbitrary score numbers (they are not arbitrary, but difficult to understand) and also make the logic more straight-forward. This first change should be a separate PR without changing the actual result and the new functionality here can be then added on top. Are you planning to take a look at this? Otherwise you can wait until we get around to do that ourselves and we'll revisit this PR once that is done.

Reply:
Thanks for your comment.
Here is the separate PR moving logic to a TextTrackScore #6178 .